sdk/src/Core/Amazon.Runtime/Internal/Auth/AWS4Signer.cs

```diff
@@ -818,26 +819,26 @@ public static string DetermineService(IClientConfig clientConfig, IRequest reque
                                                     IDictionary<string, string> pathResources,
                                                     bool doubleEncode)
         {
-            var canonicalRequest = new StringBuilder();
-            canonicalRequest.AppendFormat("{0}\n", httpMethod);
-            canonicalRequest.AppendFormat("{0}\n", AWSSDKUtils.CanonicalizeResourcePathV2(endpoint, resourcePath, doubleEncode, pathResources));
-            canonicalRequest.AppendFormat("{0}\n", canonicalQueryString);
-
-            canonicalRequest.AppendFormat("{0}\n", CanonicalizeHeaders(sortedHeaders));
-            canonicalRequest.AppendFormat("{0}\n", CanonicalizeHeaderNames(sortedHeaders));
+            var canonicalRequestBuilder = new ValueStringBuilder(512);
+            canonicalRequestBuilder.Append($"{httpMethod}\n");
```

Comment:
Does it make sense to replace such cases with
```csharp
            canonicalRequestBuilder.Append(httpMethod);
            canonicalRequestBuilder.Append('\n');
```

To avoid allocating new strings with '\n' in the end?

Reply:
I would argue given the limited number of http methods the entire span could probably be optimized further
As far as I recall I could pull off a compiler trick here. I haven't gone as far as optimizing it to that level yet because there are still some outstanding inputs from the team about why some of these methods are even protected and whether the need to continue to be so

Reply:
Addressed